aten/src/ATen/core/ivalue_inl.h

```diff
@@ -2014,6 +2014,19 @@ inline IValue::IValue(at::ArrayRef<T> v) : IValue(c10::List<T>()) {
     list.push_back(e);
   }
 }
+inline IValue::IValue(at::ArrayRef<c10::SymInt> v) : IValue() {
+  auto vi = c10::asIntArrayRefSlowOpt(v);
+  if (vi.has_value()) {
+    *this = IValue(vi);
+  } else {
+    *this = IValue(c10::List<c10::SymInt>());
```

Comment:
maybe add a comment about why we need the else (vi can be nullopt even if v has one or more elements that are symbolic)